### PR TITLE
ArchivesSpace.add_digital_object fixes, add tests

### PR DESCRIPTION
**Pull request by mistydemeo**
* Fixes an issue adding digital objects without a note specified if the parent object has notes
* Fixes an issue copying notes from the parent object if the parent object's notes do not have a label specified
* Adds tests for adding digital objects, including tests which catch the above errors

### CODE REVIEW

**Comment by Hwesta** [comment]
First commit message should be amended not to have "WIP" in the title (assuming it's not still WIP).

:+1:

**Comment by mistydemeo** [comment]
By "WIP" I meant the object being constructed, not that the commit is WIP; I'll reword it to make that clear.
